BoxContentSDK/BoxContentSDK/BOXURLSessionCacheClient.m

```diff
@@ -404,6 +404,48 @@ - (BOOL)deleteCachedInfoForUserId:(NSString *)userId associateId:(NSString *)ass
     return success;
 }
 
+- (NSDictionary *)associateIdToBackgroundSessionIdAndSessionTaskIdsForUserId:(NSString *)userId error:(NSError **)outError
+{
+    if (userId == nil) {
+        return nil;
+    }
+
+    NSError *error = nil;
+    NSArray *associateIds = [self associateIdsForUserId:userId error:&error];
+    NSMutableDictionary *associateIdToBackgroundSessionIdAndSessionTaskIds = [NSMutableDictionary new];
+
+    //iterate through users/$userId/$associateId subdirs to get its backgroundSessionId and sessionTaskId
+    for (NSString *associateId in associateIds) {
+
+        BOXURLBackgroundSessionIdAndSessionTaskId *backgroundSessionIdAndSessionTaskId = [self backgroundSessionIdAndSessionTaskIdForUserId:userId associateId:associateId error:&error];
+
+        if (backgroundSessionIdAndSessionTaskId != nil) {
+            associateIdToBackgroundSessionIdAndSessionTaskIds[associateId] = backgroundSessionIdAndSessionTaskId;
+        } else if (error == nil) {
+            //we do not have valid backgroundSessionId and sessionTaskId for a previously seen userId and associateId,
+            //something must have failed, clean up the cache dir for future usage
+            [self deleteCachedInfoForUserId:userId associateId:associateId error:&error];
```

Comment:
I wonder if in that case we should create an indicative error for 'we failed to get the info and then we failed the cleanup', and then also have one for 'we failed to get the info', because if we're in this conditional that means we don't have an error object but we did not get the item, so it is a form of failure.